src/Util/HTTPInputData.php

```diff
@@ -29,7 +29,7 @@ class HTTPInputData
 {
 	public static function process()
 	{
-		$content_parts = explode(';', $_SERVER['CONTENT_TYPE'] ?? 'application/x-www-form-urlencoded');
+		$content_parts = explode(';', static::getContentType());
```

Comment:
I never had seen `static::...` until now.

Reply:
https://www.php.net/manual/en/language.oop5.late-static-bindings.php
- `self::` ... just use static functions from inside the own class
- `static::` ... use static functions from extended classes as well (as we do)

Reply:
if we do use `self::getContentType()`, even using `HTTPInputDataDouble` won't change that it calls `HTTPInputData::getContentType()` instead of `HTTPInputDataDouble::getContentType()`

Reply:
This is a special keyword that only makes sense in inheritance in a parent class, otherwise it acts as `self::...`. Let me know if you want me to explain it to you further.